Description: Move most helper binaries to /usr/lib/codelite
Author: James Cowgill <jcowgill@debian.org>
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
diff --git a/CodeLite/cl_standard_paths.cpp b/CodeLite/cl_standard_paths.cpp
index 59ad2f93d..8527c0c02 100644
--- a/CodeLite/cl_standard_paths.cpp
+++ b/CodeLite/cl_standard_paths.cpp
@@ -113,8 +113,7 @@ wxString clStandardPaths::GetDataDir() const
 
 wxString clStandardPaths::GetBinFolder() const
 {
-    wxFileName fnExe(wxStandardPaths::Get().GetExecutablePath());
-    return fnExe.GetPath();
+    return wxStandardPaths::Get().GetPluginsDir();
 }
 
 wxString clStandardPaths::GetBinaryFullPath(const wxString& toolname) const
diff --git a/CodeLite/macros.h b/CodeLite/macros.h
index 5d00f1e7c..40fee5268 100644
--- a/CodeLite/macros.h
+++ b/CodeLite/macros.h
@@ -109,7 +109,7 @@
 #ifdef __WXGTK__
 #define TERMINAL_CMD                                               \
     wxString::Format(wxT("%s/codelite_xterm '$(TITLE)' '$(CMD)'"), \
-                     wxFileName(clStandardPaths::Get().GetExecutablePath()).GetPath().c_str())
+                     clStandardPaths::Get().GetBinFolder().c_str())
 #elif defined(__WXMAC__)
 #define TERMINAL_CMD wxString::Format(wxT("%s/OpenTerm '$(CMD)'"), clStandardPaths::Get().GetDataDir().c_str())
 #else
diff --git a/LiteEditor/app.cpp b/LiteEditor/app.cpp
index e71aae353..cf025a025 100644
--- a/LiteEditor/app.cpp
+++ b/LiteEditor/app.cpp
@@ -679,8 +679,7 @@ bool CodeLiteApp::OnInit()
 #endif
     wxString oldpath;
     wxGetEnv(wxT("PATH"), &oldpath);
-    wxFileName execfpath(clStandardPaths::Get().GetExecutablePath());
-    wxSetEnv(wxT("PATH"), oldpath + pathsep + execfpath.GetPath());
+    wxSetEnv(wxT("PATH"), oldpath + pathsep + clStandardPaths::Get().GetBinFolder());
     wxString newpath;
     wxGetEnv(wxT("PATH"), &newpath);
 
diff --git a/Plugin/pipedprocess.cpp b/Plugin/pipedprocess.cpp
index 6c5c3e05e..88f3b194b 100644
--- a/Plugin/pipedprocess.cpp
+++ b/Plugin/pipedprocess.cpp
@@ -29,6 +29,7 @@
 #include <wx/sstream.h>
 #include "wx/msgdlg.h"
 #include <wx/filename.h>
+#include "cl_standard_paths.h"
 
 PipedProcess::PipedProcess(int id, const wxString &cmdLine)
 : wxProcess(NULL, id)
@@ -55,8 +56,7 @@ void PipedProcess::Terminate()
 {
 #ifdef __WXGTK__
 	wxString cmd;
-	wxFileName exePath(clStandardPaths::Get().GetExecutablePath());
-	wxFileName script(exePath.GetPath(), wxT("codelite_kill_children"));
+	wxFileName script(clStandardPaths::Get().GetBinaryFullPath("codelite_kill_children"));
 	cmd << wxT("/bin/sh -f ") << script.GetFullPath() << wxT(" ") << GetPid();
 	wxExecute(cmd, wxEXEC_ASYNC);
 #else
